File: practice/A1.cpp
/*A1-Practical*/
/*Consider telephone book database of N clients. Make use of a hash table implementation to
quickly look up client‘s telephone number*/
#include<iostream>
#define max 10
using namespace std;


class hashTable{
private:
  int arr[max];
public:
  hashTable(){
    for(int i=0;i<max;i++){
      arr[i]=-1;
    }
  }

  int hashFunction(int entry){
    int index;
    index=entry%10;
    return index;
  }

  void push(int ent){  
    int ind,i;
    ind=hashFunction(ent);
    i=0;
    while(i<max){
      if(arr[(ind+i)%10]==-1){
        arr[(ind+i)%10]=ent;
        break;
      }
      else{
        i++;
      }
    }
  }

  void display(){
   for(int i=0;i<max;i++){  
    cout<<arr[i]<<endl;
   }             
  }
};

int main(){
  hashTable s;
  s.push(35);
  s.push(25);
  s.push(39);
  s.push(29);
  s.display();
  
  return 0;
}

File: practice/a1.py
class client:
    def_init_(self,name,phone_number):
        self.name= name
        self.phone_number=phone_number

class hashTable:
    def_init_(self,size):
        self.size=size
        for i in range(size):
            inner=[]
            for j in range(2):
                inner.append(0)
            self.append(inner)   


    def insert(self,client):
        key=client.name
        index=self.hashfunction(key)
        self.table[index].append(client)

